client/rpc-servers/src/middleware.rs

```diff
@@ -0,0 +1,117 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2017-2020 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! Middlewares for RPC servers.
+
+use http::{RequestMiddleware as HttpRequestMiddleware, RequestMiddlewareAction as HttpRequestMiddlewareAction};
+use hyper::Body;
+use prometheus_endpoint::{Registry, Counter, PrometheusError, register, U64};
+use ws::{RequestMiddleware as WSRequestMiddleware, MiddlewareAction as WSRequestMiddlewareAction};
+
+struct HTTPMetrics {
+	http_rpc_calls: Counter<U64>,
+}
+
+impl HTTPMetrics {
+	fn register(r: &Registry) -> Result<Self, PrometheusError> {
+		Ok(HTTPMetrics {
+			http_rpc_calls: register(Counter::new(
+				"rpc_http_calls",
```

Comment:
```suggestion
				"rpc_http_calls_total",
```

See https://prometheus.io/docs/practices/naming/ for details.